scaffold-api/src/scaffold_api/resources/user.py

```diff
@@ -15,69 +15,88 @@
 from flask_restx import Namespace, Resource
+from scaffold_api.services.user_service import UserService
+from scaffold_api.utils.util import cors_preflight
+from scaffold_api.schemas.user import UserSchema, UserRequestSchema
+from .apihelper import Api as ApiHelper
 
-from scaffold_api.utils.util import allowedorigins, cors_preflight
-
-API = Namespace('user', description='Endpoints for User Management')
+API = Namespace("users", description="Endpoints for User Management")
 """Custom exception messages
 """
 
+user_request_model = ApiHelper.convert_ma_schema_to_restx_model(
+    API, UserRequestSchema(), "User"
+)
+user_list_model = ApiHelper.convert_ma_schema_to_restx_model(
+    API, UserSchema(), "UserListItem"
+)
+
 
-@cors_preflight('GET, OPTIONS, POST')
-@API.route('', methods=["POST", "GET", "OPTIONS"])
+@cors_preflight("GET, OPTIONS, POST")
+@API.route("", methods=["POST", "GET", "OPTIONS"])
 class Users(Resource):
     """Resource for managing users."""
 
     @staticmethod
-    @cross_origin(origins=allowedorigins())
-    @auth.require
+    @API.response(code=200, description="Success", model=[user_list_model])
+    @ApiHelper.common_decorator(API, endpoint_description="Fetch all users")
     def get():
         """Fetch all users."""
         users = UserService.get_all_users()
-        return jsonify(users), HTTPStatus.OK
+        user_list_schema = UserSchema(many=True)
+        return user_list_schema.dump(users), HTTPStatus.OK
 
     @staticmethod
-    @cross_origin(origins=allowedorigins())
-    @auth.require
+    @ApiHelper.common_decorator(API, endpoint_description="Create a user")
+    @API.expect(user_request_model)
+    @API.response(code=201, model=user_request_model, description="UserCreated")
+    @API.response(400, "Bad Request")
     def post():
         """Create a user."""
-        user_data = request.get_json()
+        user_data = UserRequestSchema().load(API.payload)
         created_user = UserService.create_user(user_data)
-        return created_user, HTTPStatus.CREATED
+        return UserSchema().dump(created_user), HTTPStatus.CREATED
 
 
-@cors_preflight('GET, OPTIONS, PATCH, DELETE')
-@API.route('/<user_id>', methods=["PATCH", "GET", "OPTIONS", "DELETE"])
+@cors_preflight("GET, OPTIONS, PATCH, DELETE")
+@API.route("/<user_id>", methods=["PATCH", "GET", "OPTIONS", "DELETE"])
+@API.doc(params={"user_id": "The user identifier"})
 class User(Resource):
     """Resource for managing a single user"""
 
     @staticmethod
-    @cross_origin(origins=allowedorigins())
-    @auth.require
+    @ApiHelper.common_decorator(API, endpoint_description="Fetch a user by id")
+    @API.response(code=200, model=user_list_model, description="Success")
+    @API.response(404, "Not Found")
     def get(user_id):
         """Fetch a user by id."""
         user = UserService.get_user_by_id(user_id)
-        return user, HTTPStatus.OK
+        if not user:
+            return UserSchema().dump(user), HTTPStatus.NOT_FOUND
```

Comment:
if not user we still dump it?

Reply:
I will use the custom exceptions for this